rust/src/ffi.rs

```diff
@@ -593,3 +681,29 @@ pub(crate) unsafe fn get_pipeline_options(
         PipelineRetryStrategy::new(info.retry_server_error, info.retry_connection_error),
     )
 }
+
+/// FFI callback function type for PubSub messages.
+/// This callback is invoked by Rust when a PubSub message is received.
+/// The callback signature matches the C# expectations for marshaling PubSub data.
+///
+/// # Parameters
+/// * `push_kind` - The type of push notification (message, pmessage, smessage, etc.)
```

Comment:
Are there constants defined for each notification type? This comment should reference them and they should be all-caps.